### PR TITLE
Server 1296: new graph note

### PR DESCRIPTION
**Pull request by waiterZen**

### CODE REVIEW

**Comment by ar2rsawseen** [comment]
This pull request **introduces 2 alerts** when merging e3861f775e2e7449037b6556e44a41b6351d2514 into 4211496108da0ad27d2e17652445f039ca4338b8 - [view on LGTM.com](https://lgtm.com/projects/g/Countly/countly-server/rev/pr-1286039dc2d0c289e4ed2205b237c074743e49e8)

**new alerts:**

* 1 for Missing variable declaration
* 1 for Unused variable, import, function or class

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: frontend/express/public/javascripts/dom/tipsy/jquery.tipsy.js [hunk 1]
```diff
@@ -23,8 +23,26 @@
                 var $tip = this.tip();
                 
                 $tip.find('.tipsy-inner')[this.options.html ? 'html' : 'text'](title);
-                $tip[0].className = 'tipsy '+this.options.cssClass; // reset classname in case of dynamic gravity
```
Comment (ar2rsawseen) [inline frontend/express/public/javascripts/dom/tipsy/jquery.tipsy.js]: Did you recheck other places that used tipsy and if they are not broken now? 
I searched repo and some use cssClass option

File: frontend/express/public/stylesheets/main.css [hunk 2]
```diff
@@ -2841,4 +2832,380 @@ color:#E95C6C; transition:color 1s;
...
+    position: absolute;
+    z-index: 1000;
+    padding: 2px 3px 6px 3px;
+    background-color: #292929;
+    border-radius: 2px;
+    top: -25px;
+    left: 102px;
+    flex-direction: column-reverse;
+}
+.tipsy-inner {max-width: 400px !important; }
```
Comment (ar2rsawseen) [inline frontend/express/public/stylesheets/main.css]: Similar here, changing all tipsy styles would affect them in all places, not just notes

File: frontend/express/public/stylesheets/main.css [hunk 3]
```diff
@@ -2841,4 +2832,380 @@ color:#E95C6C; transition:color 1s;
...
+.tipsy-inner .note-content {min-width:150px; display: block; font-size:11px; text-align: left; font-weight: 600; max-width: 300px; margin: 10px 0px;}
+.tipsy-inner .note-content .notes-view-link { color: #2EB52B}
+.tipsy-inner .note-content .notes-view-link:hover { cursor: pointer; }
+
+.tipsy-inner .note-footer {font-size:10px; color: #D0D0D0; text-align: left;}
+.tipsy-inner .note-footer .note-owner {text-align: left; }
+.tipsy-inner .note-footer .note-type {text-align: left; height:30px; text-transform: capitalize;}
+
+
+.options-item {float: right;}
```
Comment (ar2rsawseen) [inline frontend/express/public/stylesheets/main.css]: This seems to be too broad class, can you prefix it with some container, so we know it only applies where you expect it to apply?

File: frontend/express/public/stylesheets/main.css [hunk 4]
```diff
@@ -2841,4 +2832,380 @@ color:#E95C6C; transition:color 1s;
...
+    display: inline-block;
+    text-transform: uppercase;
+}
+.note-table-view-navigator .back-link:before {
+    content: "←";
+    margin-right: 5px;
+    color: #616161;
+}
+
+.date-time-selector-container {
```
Comment (ar2rsawseen) [inline frontend/express/public/stylesheets/main.css]: This seems to be too broad class, can you prefix it with some container, so we know it only applies where you expect it to apply?

File: frontend/express/public/javascripts/dom/tipsy/jquery.tipsy.js (hunk 1 above)
Comment (waiterZen) [reply frontend/express/public/javascripts/dom/tipsy/jquery.tipsy.js]: ok, let me resume the customized class name to tipsy view.

File: api/parts/mgmt/apps.js [hunk 5]
```diff
@@ -738,10 +738,6 @@ function deleteAllAppData(appId, fromAppDelete, params, app) {
             }, deleteEvents);
         }
     });
-
-    if (fromAppDelete) {
-        common.db.collection('graph_notes').remove({'_id': common.db.ObjectID(appId)}, function() {});
-    }
```
[the remark this answers is not in the document]
Comment (waiterZen) [reply api/parts/mgmt/apps.js]: fixed, new graph_notes collection name is "notes".

File: frontend/express/public/stylesheets/main.css (hunk 2 above)
Comment (waiterZen) [reply frontend/express/public/stylesheets/main.css]: solved

**Pull request merged**